API documentation.md

```diff
@@ -1,60 +1,169 @@
 
-Collection description
+A collection of documents representing key nodes in the relocation process of Kiruna. Each document corresponds to agreements, conflicts, consultations, material effects, or other documents that directly influence the project. Documents are categorized by type, including informative, prescriptive, and original resources, and are connected through a network of direct and collateral consequences, projections, and updates.
 
-## GET /template/{id}
+## GET `/documents`
 
-Endpoint description
+Retrieve all documents.
 
 ### Query parameters
 
-- `filter1` (optional): filter 1 description;
-- `filter2` (optional): filter 2 description.
+TODO: filters
+
+## Response body
+
+```json
+[
+    {
+        "id": 1,
+        "title": "",
+        "description": "",
+        "coordinates": {
+            "latitude": "",
+            "longitude": ""
+        },
+        "links": [
+            {
+                "targetNodeId": 2,
+                "linkType": "DIRECT"
+            },
+            ...
+        ],
+        //other properties
+    },
+    ...
+]
+```
+
+### Errors
+
+This API can return the following error codes:
+
+- `401 Unauthorized`: You are unauthorized.
+- `500 Internal Server Error`: An unexpected error occurred on the server.
+
+## GET `/documents/{id}`
+
+Retrieve a specific document by its unique identifier.
+
+### Query parameters
+
+TODO:
 
 ### Response body
 
 ```json
 {
-    "attr1": "attr1",
-    "attr2": "attr2"
+    "id": 1,
+    "title": "",
+    "description": "",
+    "coordinates": {
+        "latitude": "",
+        "longitude": ""
+    },
+    "links": [
+        {
+            "targetNodeId": 2,
+            "linkType": "DIRECT"
+        },
+        ...
+    ],
+    //other properties
 }
 ```
 
 ### Errors
 
 This API can return the following error codes:
 
+- `401 Unauthorized`: You are unauthorized.
+- `404 Not Found`: The requested document was not found.
+- `500 Internal Server Error`: An unexpected error occurred on the server.
+
+## POST `/documents`
+
+Create a new document.
+
+### Request body
+
+```json
+{
+    "id": 1,
+    "title": "",
+    "description": "",
+    "coordinates": {
+        "latitude": "",
+        "longitude": ""
+    },
+    //other properties
+}
+```
+
+### Response body
+
+```json
+{
+    "message": "Document created successfully",
+    "id": 1
+}
+```
+
+### Errors
+
+This API uses the following error codes:
+
 - `400 Bad Request`: The request was malformed or missing required parameters.
-- `401 Unauthorized`: The API key provided was invalid or missing.
-- `404 Not Found`: The requested resource was not found.
+- `401 Unauthorized`: You are unauthorized.
 - `500 Internal Server Error`: An unexpected error occurred on the server.
 
-## POST /template
+## PATCH `/documents/{id}`
 
-Endpoint description
+Update an existing document by its unique identifier.
 
 ### Request body
 
 ```json
 {
-    "attr1": "random1",
-    "attr2": "random2"
+    "id": 1,
```

Comment:
Remove id, it's already in the query URL

Reply:
Done